imxvpuapi/imxvpuapi_vpulib.c

```diff
@@ -2629,10 +2644,10 @@ static void imx_vpu_enc_set_mjpeg_tables(unsigned int quality_factor, EncMjpgPar
 	else
 		scale_factor = 200 - quality_factor * 2;
 
-	imx_vpu_enc_copy_quantization_table(mjpeg_params->qMatTab[DC_TABLE_INDEX0], mjpeg_enc_quantization_luma,   sizeof(mjpeg_enc_quantization_luma),   scale_factor);
```

Comment:
Ah, this is an interesting. It does contradict the indices from the imx-vpu vpu_lib.h header:

    #define DC_TABLE_INDEX0		    0
    #define AC_TABLE_INDEX0		    1
    #define DC_TABLE_INDEX1		    2
    #define AC_TABLE_INDEX1		    3

but these indices themselves do not seem to apply for MJPEG. From looking at the JpgDecQMatTabSetUp() function inside imx-vpu's vpu_util.c function, your change seems to be correct. Is this how you've noticed that the order was wrong?